src/Common/src/System/Net/SafeCloseSocket.Windows.cs

```diff
@@ -33,52 +33,56 @@ public ThreadPoolBoundHandle IOCPBoundHandle
         // Binds the Socket Win32 Handle to the ThreadPool's CompletionPort.
         public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCompletionPortOnSuccess)
         {
+            // Check to see if the socket native _handle is already
+            // bound to the ThreadPool's completion port.
+            if (_released || _iocpBoundHandle == null)
```

Comment:
Maybe with other refactorings or changes, but at present it doesn't look like it... _iocpBoundHandle starts out as null and is then lazily initialized to non-null, after which point it's never set back to null.  From the comments in the code, it looks like this is done to support freeing a NativeOverlapped after disposal.